src/compile/legend/parse.ts

```diff
@@ -86,17 +86,20 @@ function isExplicit<T extends string | number | object | boolean>(
   fieldDef: TypedFieldDef<string>
 ) {
   switch (property) {
+    case 'disable':
+      return legend !== undefined; // if axis is specified or null/false, then it's enable/disable state is explicit
+
     case 'values':
       // specified legend.values is already respected, but may get transformed.
-      return !!legend.values;
+      return !!legend?.values;
     case 'title':
       // title can be explicit if fieldDef.title is set
       if (property === 'title' && value === fieldDef.title) {
         return true;
       }
   }
   // Otherwise, things are explicit if the returned value matches the specified property
-  return value === legend[property];
+  return value === (legend || {})[property];
```

Comment:
```suggestion
  return value === (legend ?? {})[property];
```

Reply:
why is that better?  When it produces the same result, `||` is probably better performance and leads to simpler JS output  code.